pkg/motor/bucket_resolver.go

```diff
@@ -72,32 +63,20 @@ func (mtr *motorNodeImpl) GetBucket(context context.Context, did string) (bucket
 	return mtr.Resources.bucketStore[did], nil
 }
 
+/*
+	Takes the whereIs store and checks for a matching bucket in the cache, if its not present it will create it and get its sub buckets
+	Does not query for new buckets, only respects what is currently present in the store
```

Comment:
This should probably query instead of creating. If someone is calling "Get" they won't expect new resources to be created